phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1273,6 +1387,121 @@ public void doMutation() throws IOException {
+     *                                 were already sent and those that were unsent
+     * @param unsentMutationBatchList list of mutation batches that are unsent
+     * @param numMutations total number of mutations
+     * @param numFailedMutations number of failed mutations in the most recent failed batch
+     * @param numFailedPhase3Mutations number of mutations failed in phase 3 of index commits
+     * @param mutationCommitTime time taken for committing all mutations
+     * @return mutation metric object just accounting for mutations that are already
+     * successfully committed
+     */
+    @VisibleForTesting
+    static MutationMetric getCommittedMutationsMetric(
+            MutationBytes totalMutationBytesObject, List<List<Mutation>> unsentMutationBatchList,
+            long numMutations, long numFailedMutations,
+            long numFailedPhase3Mutations, long mutationCommitTime) {
+        long committedUpsertMutationBytes = totalMutationBytesObject == null ? 0 :
+                totalMutationBytesObject.getUpsertMutationBytes();
+        long committedDeleteMutationBytes = totalMutationBytesObject == null ? 0 :
+                totalMutationBytesObject.getDeleteMutationBytes();
+        long committedUpsertMutationCounter = totalMutationBytesObject == null ? 0 :
+                totalMutationBytesObject.getUpsertMutationCounter();
+        long committedDeleteMutationCounter = totalMutationBytesObject == null ? 0 :
+                totalMutationBytesObject.getDeleteMutationCounter();
+        long committedTotalMutationBytes = totalMutationBytesObject == null ? 0 :
+                totalMutationBytesObject.getTotalMutationBytes();
+        long upsertMutationCommitTime = 0L;
+        long deleteMutationCommitTime = 0L;
+
+        if (totalMutationBytesObject != null && numFailedMutations != 0) {
+            List<Mutation> uncommittedMutationsList = new ArrayList<>();
+            for (List<Mutation> mutationBatch : unsentMutationBatchList) {
+                uncommittedMutationsList.addAll(mutationBatch);
+            }
+            // Calculate the uncommitted mutations
+            MutationBytes uncommittedMutationBytesObject =
+                    calculateMutationSize(uncommittedMutationsList, false);
+            committedUpsertMutationBytes -=
+                    uncommittedMutationBytesObject.getUpsertMutationBytes();
+            committedDeleteMutationBytes -=
+                    uncommittedMutationBytesObject.getDeleteMutationBytes();
+            committedUpsertMutationCounter -=
+                    uncommittedMutationBytesObject.getUpsertMutationCounter();
+            committedDeleteMutationCounter -=
+                    uncommittedMutationBytesObject.getDeleteMutationCounter();
+            committedTotalMutationBytes -=
+                    uncommittedMutationBytesObject.getTotalMutationBytes();
+        }
+
+        // TODO: For V1, we don't expect mixed upserts and deletes so this is fine,
```

Comment:
Remove this comment or modify it to explain what should be covered in "V1" version of this feature

Reply:
Removed it